### PR TITLE
[adobe_analytics] Support for Identity's getExperienceCloudId

### PR DESCRIPTION
**Pull request by jeremie-movify**

### CODE REVIEW

File: packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/Extractor.kt [hunk 1]
```diff
@@ -5,12 +5,14 @@ import io.flutter.plugin.common.MethodCall
 object Extractor {
     private enum class AdobePluginCall(val rawMethodName: String? = null) {
         TRACK("track"),
+        GETEXPERIENCECLOUDID("getExperienceCloudId"),
```
Comment (timrijckaert) [inline packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/Extractor.kt]: `GET_EXPERIENCE_CLOUD_ID`

File: packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt [hunk 2]
```diff
@@ -22,6 +24,16 @@ class AdobeAnalyticsPlugin : MethodCallHandler {
         val exhaustMe: Any = when (val data = Extractor.adobeCallFromCall(call)) {
             is Extractor.AdobeCall.TrackState -> trackState(data, result)
             is Extractor.AdobeCall.TrackAction -> trackAction(data, result)
+            is Extractor.AdobeCall.GetExperienceCloudId -> {
```
Comment (timrijckaert) [inline packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt:30]: `Looper.getMainLooper()`

**Review by timrijckaert — CHANGES_REQUESTED** [review]

File: packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt [hunk 3]
```diff
@@ -22,6 +24,16 @@ class AdobeAnalyticsPlugin : MethodCallHandler {
...
+                val mainLooper = Handler()
+                Thread().run {
```
Comment (timrijckaert) [inline packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt]: Not sure how this `Thread` will be GC'd after use.... 🤔

**Review by timrijckaert — CHANGES_REQUESTED** [review]

File: packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/Extractor.kt (hunk 1 above)
Comment (jeremie-movify) [reply packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/Extractor.kt]: Updated 👍

File: packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt (hunk 2 above)
Comment (jeremie-movify) [reply packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt:30]: PR updated ; we now use `Executors.newCachedThreadPool()`

File: packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt (hunk 3 above)
Comment (jeremie-movify) [reply packages/adobe_analytics/android/src/main/kotlin/be/vrt/adobe_analytics/AdobeAnalyticsPlugin.kt]: Verified GC with @petermnt  ; works as intended 👍

**Review by timrijckaert — APPROVED** [review]

**Review by Kymer — APPROVED** [review]

**Review by Kymer — APPROVED** [review]

**Review by timrijckaert — APPROVED** [review]

**Pull request merged**